src/main/java/application/rest/v1/configmaps/KindActionMappingCache.java

```diff
@@ -0,0 +1,248 @@
+
+            @Override
+            public String getWatcherThreadName() {
+                return WATCHER_THREAD_NAME;
+            }
+
+            @Override
+            public List<Object> listResources(ApiClient client, AtomicReference<String> resourceVersion)
+                    throws ApiException {
+                String methodName = "listResources";
+                final CustomObjectsApi coa = new CustomObjectsApi();
+                coa.setApiClient(client);
+                if (Logger.isDebugEnabled()) {
+                    Logger.log(CLASS_NAME, methodName, Logger.LogType.DEBUG,
+                            "\n List KAM Custom Resources for all namespaces with" + "\n group = " + 
+                            KAM_GROUP + "\n version = " + KAM_VERSION + "\n plural = " + KAM_PLURAL);
+                }
+
+                Object kamCRs = coa.listClusterCustomObject(KAM_GROUP, KAM_VERSION, KAM_PLURAL, null, null, null, null);
+                return Watcher.processCustomObjectsApiList(client, kamCRs, resourceVersion);
+            }
+
+            @Override
+            public Call createWatchCall(ApiClient client, String resourceVersion) throws ApiException {
+                final CustomObjectsApi coa = new CustomObjectsApi();
+                coa.setApiClient(client);
+                return coa.listClusterCustomObjectCall(KAM_GROUP, KAM_VERSION, KAM_PLURAL, null, null, resourceVersion,
+                        Boolean.TRUE, null, null);
+            }
+
+            @SuppressWarnings("serial")
+            @Override
+            public Type getWatchType() {
+                return new TypeToken<Watch.Response<Object>>() {
+                }.getType();
+            }
+
+            @Override
+            public void processResponse(ApiClient client, String type, Object object) {
+                Object kamWatcher = KAM_WATCHER_REF.get();
+                if (kamWatcher== null) {
+                    updateModCount(); // Prevents a stale cached list from being returned when the list is restored.
+                    kamWatcher = new Object();
+                    KAM_WATCHER_REF.set(kamWatcher);
+                    updateModCount();
+                }
+
+                switch (type) {
+                    case "ADDED":
+                    case "MODIFIED":
+                    case "DELETED":
+                        if (Logger.isDebugEnabled()) 
+                            Logger.log(getClass().getName(), "processResponse", Logger.LogType.DEBUG, 
+                                "KindActionMapping CRs are being add/modified/deleted while the kam watcher is running so update mod count");
+                        updateModCount();
+                        break;
+                }
+            }
+
+            @Override
+            public void reset(ApiClient client) {
+                // If the watch stops or fails delete the cache.
+                KAM_WATCHER_REF.set(null);
+                //KAM_CACHE.set(null);
```

Comment:
`KAM_CACHE_REF.set(null);` here? It may work without it but can help prevent a stale list from being used.